### ISSUE TITLE
false positive: no-shadowed-variable

### ISSUE THREAD

**Issue by unional** [palantir/tslint#1099]
tslint: 3.7.1

```ts
export default function applyMixins(
    derivedCtor: new (...args: any[]) => any,
    baseCtors: Array<new (...args: any[]) => any>): void {
    // ...
    });
};
```
error: shadowed variable: 'args'

**Comment by bencoveney** [comment]
Have another set of reproduction steps if it helps. Presumably the same issue:

```typescript
class MyClass { };

declare type MyConstructor = new (myVariable: any) => MyClass;

function MyFunction(constructor: MyConstructor)
{
	const myVariable = 1;
	console.log(myVariable);
}
```

**Comment by adidahiya** [comment]
@bencoveney which identifier is reported as shadowed in that example?

**Comment by bencoveney** [comment]
Sorry, the error message is:

`Shadowed variable: 'myVariable'`

**Comment by IllusionMH** [comment]
@adidahiya, @JKillian looks like parameters from `ConstrustorType` and `FunctionType` nodes shouldn't be added to scope. What do you think on this kind of fix?

**Comment by adidahiya** [comment]
@IllusionMH sounds reasonable. I assume that those syntax kinds are distinct from constructor/function declarations? i.e. this would still produce a lint failure:

```ts
function MyFunction(x: any) {
  let x: string;
      ^ Shadowed variable 'x'
}
```

**Comment by JKillian** [comment]
Fixed by #1482
